src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/ScenarioRunExtensions.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI.Evaluation.Reporting;
+
+/// <summary>
+/// Extension methods for <see cref="ScenarioRun"/>.
+/// </summary>
+public static class ScenarioRunExtensions
+{
+    private static IEnumerable<ChatMessage> EmptyMessages { get; } = [];
```

Comment:
Do you need this, or can you just use `[]` wherever you need it?

Reply:
I was not sure whether `[]` would result in a new instance being allocated each time... But this may be a case of premature optimization even if this does allocate... I've gone ahead and replaced all usages of `EmptyMessages` with `[]`.

Reply:
`[]` target-typed to `IEnumerable<T>` expands by the compiler to `Array.Empty<T>()`.